vortex-scalar/src/utf8.rs

```diff
@@ -11,7 +11,7 @@ use crate::{InnerScalarValue, Scalar, ScalarValue};
 #[derive(Debug, Hash)]
 pub struct Utf8Scalar<'a> {
     dtype: &'a DType,
-    value: Option<BufferString>,
+    value: Option<Arc<BufferString>>,
```

Comment:
Why have the arc here? It only exists in the ScalarValue enum to minimise the size of the struct. BufferString is still internally zero-copy